react/src/components/Timeline/TimelineContainer.tsx

```diff
@@ -0,0 +1,55 @@
+import { Box, Flex, Text } from "@chakra-ui/react";
+import { User } from "types/generated/graphql";
+import TimelineEntry from "./components/TimelineEntry";
+
+export interface ITimelineEntry {
+  action: string | undefined;
```

Comment:
I'd suggest try to get rid of these undefined items here and handle these cases before you pass the data in the component. In general, that would be just cleaner.